src/cogent3/core/profile.py

```diff
@@ -292,6 +293,53 @@ def entropy(self):
         result = self.entropy_terms().row_sum()
         return result.array
 
+    def relative_entropy_terms(self, background=None):
+        """
+        Parameters
+        ----------
+        background : dict
+            {motif_1: prob_1, motif_2: prob_2, ...} is the specified background distribution.
+
+        Returns
+        -------
+        self.template.wrap(ret) : DictArray
+            The term wise relative entropy
+
+        Notes
+        -----
+        If background is type None, it defaults to equifrequent.
+
+        """
+        if background is None:
+            num_motifs = len(self.motifs)
+            background = array([1 / num_motifs] * num_motifs)
+        else:
+            background = array([background.get(m, 0) for m in self.motifs])
+
+        assert_allclose(background.sum(), 1.0)
```

Comment:
please replace this assert line by calling `validate_freqs_array` which is in profile already (I had forgotten this function existed).